Meadowlark-js/packages/meadowlark-core/src/validation/PaginationValidator.ts

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+/* eslint-disable no-use-before-define */ // function declarations are hoisted properly in ES6
+import { PaginationParameters } from '../message/PaginationParameters';
+import { createInvalidRequestResponse } from '../Utility';
+
+const isNotPositiveInteger = (value: string): Boolean => !Number(value) || Number.parseInt(value, 10) < 1;
+
+/**
+ * Validates the `limit` and `offset` parameters from a query string.
+ */
+export function validatePaginationParameters(parameters: PaginationParameters): string | undefined {
+  // Acceptable for neither to be defined
+  if (parameters.limit == null && parameters.offset == null) {
+    return undefined;
+  }
+
+  const limit: string[] = [];
+  const offset: string[] = [];
+
```

Comment:
I see where you are coming from, but if I do that, then I'll have to rename the properties when creating the HTTP response message, which should be of the form similar to the ODS/API:

```
{
    "message": "The request is invalid.",
    "modelState": {
        "request.Limit": [
            "The value 'a' is not valid for Limit."
        ]
    }
}
```

Trying to match the ODS/API error messages is a new idea that came up from the SIG. We'll need to change other validation processes in a future ticket. I did not try to match precisely at this point, because "request.Limit" feels too much like an artifact of .NET.